### ISSUE TITLE
Save completed atomic data entry data for a product to its ticket

### ISSUE THREAD

**Issue by ChrisKellaleaMaynard** [aehrc/snomio#290]
**As a:** Terminology Author,  
**I want to:**  have the information I've entered into the atomic data entry screen to be automatically saved to its ticket when I submit the information to create concepts within Snomio
**So I can:**  retain a record of the atomic data for auditing purposes, or to load a new atomic data entry screen if necessary.

---
### Related Issues
* #2 
* #35 
* #51 
* #91 
* #110 
* #192 
* #269 

### Acceptance Criteria
1. Once sufficient information has been entered to support the creation of concepts, the _Submit_ button (or equivalent) becomes enabled.
2. If the user makes further changes which invalidates the ability to create concepts (e.g. removes information which was previously added) the _Submit_ button becomes disabled again.
3. If the user clicks the _Submit_ button then the final version of the data is saved against the ticket as a JSON file attachment.
4. The data is saved for the product being entered, to support multiple products being created/modified within the one ticket.
5. The JSON file's filename has the following format: **ProductData-<_CTPP Preferred Term_>-<_timestamp, as YYYYMMDD_>(<1-based index, if atomic data has already been saved for the same _CTPP Preferred Term_/timestamp combination).json**
6. The atomic data is saved in formatted JSON, which is able to be easily read from the ticket programmatically so it can be used to repopulate the atomic data entry screen in its original state.  A separate story covers the rendering of this data in a more user-friendly format.
7. Changes are not automatically saved when the user switches to another browser window/tab (e.g. when displaying a ticket's details, or transferring to the Authoring Platform).
8. If an error is encountered while saving the changes, the user is warned and the error details are displayed.

### Testing
1. [ test case ]

**Comment by senjo-aehrc** [comment]
@dionmcm ,Can you please update this ticket once the device creation api is ready

**Comment by ChrisKellaleaMaynard** [comment]
@senjo-aehrc  and @dionmcm  this ticket has been modified to restrict its scope to only medicine products.  Devices are now the scope of issue #425 to allow 290 to proceed while the device-saving issues are resolved.

**Comment by ChrisKellaleaMaynard** [comment]
Review passed, although errors were encountered with validation - captured in ticket #460 .
